### ISSUE TITLE
[BUG] EventProcessorClient - Can't create session when the connection is closing. 

### ISSUE THREAD

**Issue by TarekAljabban** [Azure/azure-sdk-for-net#13745]
I'm using `EventProcessorClient` in my service to monitor multiple EventHubs. I register error handler with my client using `EventProcessorClient.ProcessErrorAsync += EventProcessor.ProcessErrorAsync;`
In ProcessErrorAsync method, I print the content of `ProcessErrorEventArgs errorEventArgs`.

Occasionally, `ProcessErrorAsync` is invoked for a specific EventHub (different one every time). With exception message: "Can't create session when the connection is closing."
Once this exception occurs for one EventHub, it keeps happening periodically on that same EventHub.

I have not noticed any service degradation, and events seem to still be picked up and processed successfully but I'm not sure about this part since I don't have rigorous tests for my service yet.

**Exception Details**

**Message**
Can't create session when the connection is closing. 

**Operation**
Reading events from the Event Hubs service. 

**Exception Call stack:**
```
System.InvalidOperationException:
   at Azure.Messaging.EventHubs.Consumer.EventHubConsumerClient+<>c__DisplayClass46_0+<<PublishPartitionEventsToChannelAsync>g__performCleanup|0>d.MoveNext (Azure.Messaging.EventHubs, Version=5.0.1.0, Culture=neutral, PublicKeyToken=92742159e12e44c8)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Azure.Messaging.EventHubs.Consumer.EventHubConsumerClient+<ReadEventsFromPartitionAsync>d__37.MoveNext (Azure.Messaging.EventHubs, Version=5.0.1.0, Culture=neutral, PublicKeyToken=92742159e12e44c8)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Threading.Tasks.Sources.ManualResetValueTaskSourceCore`1.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Azure.Messaging.EventHubs.EventProcessorClient+<>c__DisplayClass94_0+<<RunPartitionProcessingAsync>b__0>d.MoveNext (Azure.Messaging.EventHubs.Processor, Version=5.0.1.0, Culture=neutral, PublicKeyToken=92742159e12e44c8)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Azure.Messaging.EventHubs.EventProcessorClient+<>c__DisplayClass94_0+<<RunPartitionProcessingAsync>b__0>d.MoveNext (Azure.Messaging.EventHubs.Processor, Version=5.0.1.0, Culture=neutral, PublicKeyToken=92742159e12e44c8)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Azure.Messaging.EventHubs.EventProcessorClient+<>c__DisplayClass94_0+<<RunPartitionProcessingAsync>b__0>d.MoveNext (Azure.Messaging.EventHubs.Processor, Version=5.0.1.0, Culture=neutral, PublicKeyToken=92742159e12e44c8)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Azure.Messaging.EventHubs.EventProcessorClient+<>c__DisplayClass94_0+<<RunPartitionProcessingAsync>b__0>d.MoveNext (Azure.Messaging.EventHubs.Processor, Version=5.0.1.0, Culture=neutral, PublicKeyToken=92742159e12e44c8)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Azure.Messaging.EventHubs.EventProcessorClient+<StopPartitionProcessingIfRunningAsync>d__102.MoveNext (Azure.Messaging.EventHubs.Processor, Version=5.0.1.0, Culture=neutral, PublicKeyToken=92742159e12e44c8)
```

**Environment:**
 - Assembly Azure.Messaging.EventHubs.Processor, Version=5.0.1.0
 - Service running in Azure VMSS.

**Comment by jsquire** [comment]
Hi @TarekAljabban.  Thanks for reaching out, and we regret that you're experiencing difficulties.  This is a known issue that was fixed for v5.1.0 of the library.   The newer version of the library is backwards compatible without breaking changes; if you're able to upgrade, I'd recommend doing so as we refined the `EventProcessorClient` quite a bit in that release.

The underlying issue that you're seeing is a race condition that occurs when the AMQP link to the service was disconnected but the protocol library had not finished cleaning up when it began reconnecting.  This should not impact the operation of the processor in terms of resilience or performance, but will cause some noise in logs.
